File: Python/Excercise/Exercise 3. String Indexing and Even Slicing.py
"""
    Date: 10-07-2026

    Exercise 3. String Indexing and Even Slicing
    Practice Problem: Display only those characters which are present at an even index number in given string.

    Exercise Purpose: Understand how data is stored in memory using zero-based indexing. In most languages, the first character is at position 0, the second at 1, and so on. Mastering indexing is vital for data parsing.

    Given Input: String: "pynative"

    Expected Output:

    Original String is  pynative
    Printing only even index chars
    p
    n
    t
    v
"""

str = "pynative"

print("Original String is pynative\nPrinting only even index chars")
i = 0
while i < len(str):
    if (i % 2 == 0):
        print(str[i])
    i = i + 1

File: Java Script/Exercises/Exercise 3. String Indexing and Even Slicing.js
const str = "JavaScript";

console.log("Original String is", str);
console.log("Printing only even index chars");

console.log("Method 1");
for (let i = 0; i < str.length; i++){
    if(i % 2 == 0){
        console.log(str[i])
    }
}

console.log("Method 2");
for (let i = 0; i < str.length; i += 2){
    console.log(str[i]);
}